tinygrad/nn/state.py

```diff
@@ -1,11 +1,25 @@
-import os, json, pathlib, zipfile, pickle, tarfile, struct, functools
-from typing import Dict, Union, List, Optional, Any, Tuple, Callable
+import os, json, pathlib, zipfile, pickle, tarfile, struct, functools, io
+from typing import Dict, Union, List, Optional, Any, Tuple, Callable, BinaryIO
 from tinygrad.tensor import Tensor
 from tinygrad.dtype import dtypes
 from tinygrad.helpers import prod, argsort, DEBUG, Timing, CI, unwrap, GlobalCounters, tqdm
 from tinygrad.shape.view import strides_for_shape
 from tinygrad.multi import MultiLazyBuffer
 
+class RawTensorIO(io.RawIOBase, BinaryIO): # type: ignore[misc]  # incompatible definitions of methods in the base classes
+  def __init__(self, t: Tensor):
+    if len(t.shape) != 1 or t.dtype != dtypes.uint8: raise ValueError("Invalid Tensor, expected 1 dimensional tensor with dtype=uint8")
+    self._position, self._tensor = 0, t
+  def readable(self) -> bool: return True
+  def readinto(self, buffer: Union[bytearray, memoryview]) -> int:  # type: ignore[override]  # expects Buffer, which is not always writable
+    data = self._tensor[self._position:self._position+len(buffer)].data()
+    buffer[:len(data)], self._position = data, self._position + len(data)
```

Comment:
don't put this on one line, it's harder to read

Reply:
Will split into two lines.